### ISSUE TITLE
Nested multi-line comments not highlighted correctly

### ISSUE THREAD

**Issue by ahelwer** [alygin/vscode-tlaplus#230]
Here's the code:
```tla
---- MODULE test ----
(*this
is
a
(*
nested
*)
multi
line
comment
*)
====
```
this is highlighted as:
![image](https://user-images.githubusercontent.com/718765/124995467-6ea25700-e015-11eb-87bf-1555e2ef8b44.png)
I am not even sure this is theoretically parse-able with regular expressions lol.

**Comment by alygin** [comment]
@ahelwer, you're right, TextMate grammar is not powerful enough to highlight such things. It's only possible with a full-featured parser. 

Maybe your tree-sitter grammar will make it to a TLA+ LSP sometime :) That would make highlighting much more accurate, let alone other cool features.

**Comment by ahelwer** [comment]
@alygin did you fork https://github.com/agentultra/TLAGrammar (same as what github uses) or write your own textmate grammar for highlighting? I don't know whether the github grammar is really being maintained.

**Comment by ahelwer** [comment]
@alygin check it out! Progress https://tlaplus-community.github.io/tree-sitter-tlaplus/
